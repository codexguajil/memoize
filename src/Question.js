import React, { Component } from 'react';
import Answer from './Answer';

class Question extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="question" id="Question">
        <h3>{this.props.question.question}</h3>
        {
          this.props.question.allAnswers.map((answer,i) => {
            return <Answer key={i}
                           answer={answer}
                           correct={this.props.question.correctAnswer}
                           toastLaunch={this.props.launchToast}
                           classes={this.props.classes}
                    />
          })
        }
        
      </div>
    )
  }
}

export default Question;