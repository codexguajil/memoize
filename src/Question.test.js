import React from 'react';
import Question from './Question';
import { shallow } from 'enzyme';

describe("Question", () => {

  let wrapper;
  let mockFunction = jest.fn();

  let mockQuestions = [
    {
    "id": 1,
    "question": "What does OOP stand for?",
    "correctAnswer": "Object Oriented Programming",
    "allAnswers": [
      "Object Oriented Poop",
      "Only Ogre Pants",
      "Object Oriented Programming",
      "Object Orientation Painstaking"
    ],
    "category": "oop"
  },
  {
    "id": 2,
    "question": "Which terminal command creates a new directory?",
    "correctAnswer": "mkdir",
    "allAnswers": [
      "mv",
      "cd",
      "mkdir",
      "git init"
    ],
    "category": "terminal"
  }
  ]

  let mockIndex = 0;

  beforeEach(() => {
    wrapper = shallow (
        <Question launchToast={mockFunction} 
                  changeQuestion={mockFunction} 
                  question={mockQuestions[mockIndex]}
                  correctAnswer={mockQuestions[mockIndex].correctAnswer}
        />
    )
  })


  it("should render the question", () => {
    expect(wrapper).toMatchSnapshot()
  })
})