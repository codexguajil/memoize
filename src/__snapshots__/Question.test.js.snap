// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should render the question 1`] = `
<div
  className="question"
  id="Question"
>
  <h3>
    What does OOP stand for?
  </h3>
  <Answer
    answer="Object Oriented Poop"
    correct="Object Oriented Programming"
    key="0"
    toastLaunch={[MockFunction]}
  />
  <Answer
    answer="Only Ogre Pants"
    correct="Object Oriented Programming"
    key="1"
    toastLaunch={[MockFunction]}
  />
  <Answer
    answer="Object Oriented Programming"
    correct="Object Oriented Programming"
    key="2"
    toastLaunch={[MockFunction]}
  />
  <Answer
    answer="Object Orientation Painstaking"
    correct="Object Oriented Programming"
    key="3"
    toastLaunch={[MockFunction]}
  />
</div>
`;
